### ISSUE TITLE
False positives when importing from flow-typed module declarations

### ISSUE THREAD

**Issue by dlindenkreuz** [benmosher/eslint-plugin-import#708]
Explicit imports from [flow-typed](https://github.com/flowtype/flow-typed) module declarations seem to yield false positives.

Example:

```js
import type { ActionCreators } from "redux"
// error  ActionCreators not found in 'redux'  import/named
```

The error should not appear because [flow can attach declared types to named modules via flow-typed or other library files](https://flowtype.org/docs/third-party.html#using-flow-typed), even if the modules themselves do not export those types. In the example above, redux does not export any types. Instead, a [flow-typed library](https://github.com/flowtype/flow-typed/blob/master/definitions/npm/redux_v3.x.x/flow_v0.33.x-/redux_v3.x.x.js) handles the type declarations.

**Comment by skovhus** [comment]
Seeing the same error. Not sure exactly how to solve it.

**Comment by mjomble** [comment]
I guess there are two main different ways to solve this problem:

1. Try to replicate all of Flow's import logic in this rule and keep it in sync with any future changes.
2. Ignore all `import type` declarations in this rule.

I'm leaning towards option 2.

If the user is already using Flow, there's a good chance Flow itself will catch any invalid imports like this.

In most cases, there should be no need for two different tools reporting the same problem, one would be enough. Often even preferred.

If you want, you can apply a temporary local hack for the second solution by editing `node_modules/eslint-plugin-import/lib/rules/named.js` and adding

```javascript
if (node.importKind === 'type') return;
```

to the start of `checkSpecifiers`.
This will be gone and require reapplying after each `eslint-plugin-import` update, though.

**Comment by petegleeson** [comment]
I feel as though option 2 would be a good solution to this problem. @mjomble you are right that flow will catch any invalid imports. Is anyone working on this? Otherwise I might see if I can add this as a rule to the plugin.

**Comment by athomann** [comment]
Experiencing the same thing.   Anyone working on a PR for this? I may be able to help.

**Comment by mjomble** [comment]
I'm not working on a PR

**Comment by SpainTrain** [comment]
From our experience and comments/reactions in https://github.com/benmosher/eslint-plugin-import/pull/881, it seems like this issue may still be open.  Possible to reopen, or should I open a new issue?  Thanks!

**Comment by ljharb** [comment]
Please open a new issue; it's been many months, so it's possible your problem is different.
